scene/main/http_request.cpp

```diff
@@ -349,6 +349,9 @@ bool HTTPRequest::_update_connection() {
 			}
 
 			client->poll();
+			if (client->get_status() != HTTPClient::STATUS_BODY) {
+				break; // State changed after this poll, will check at next iteration.
```

Comment:
Please don't hate me, since I originally suggested the change, but this should actually be: `return false`, otherwise it will still work, but print an error in the debugger since it will reach the `ERR_FAIL_V(false)` at the end of the switch case. :heart:

Reply:
Haha, No issues <3! i don't even fully remember what this was about! Since we're reviving all this i think its worth pointing out that partly because of the project I'm working with has complicated networking needs and also partly because i found networking in Godot to be _not so reliable_  when messing around with framerates (probably because of networking depending on polling through the main loop). I remember for example having trouble with web sockets losing packets when i set the framerate to 1 (My project is an application so i have to set the frame rates to 1 when its out of focus). I moved on to using SignalR with C# and also experimenting with mqtt (which is what i'm probably going to use). It would be interesting to see an MQTT implementation for Godot though. MQTT is an interesting technology mostly used in IoT but it could probably be used for anything. I'm a bit tight on time right now since i have to deliver the project in the next 3 months but i will consider working on an MQTT implementation for Godot in the near future! Sorry for going out of topic there, i will make the changes and push!